app/models/subscription.rb

```diff
@@ -0,0 +1,19 @@
+class Subscription < ApplicationRecord
+  enum status: %i[active trashed]
+  enum frequency: %i[daily]
+
+  validates :email, email_address: { presence: true }
+  validates :frequency, presence: true
+  validates :search_criteria, uniqueness: { scope: %i[email expires_on frequency] }
+
+  scope :ongoing, -> { active.where('expires_on >= current_date') }
+
+  before_save :set_reference
```

Comment:
Active record hooks make be wary as they can be called at unexpected times. I found some considerations [1] which I think are worth considering. 

- should this run with `after_commit` instead of `after_save` or `before_save`
- do we need to set this after save at all? could we set it on the initialisation, or are there other considerations to be made? 

https://www.justinweiss.com/articles/a-couple-callback-gotchas-and-a-rails-5-fix/

Reply:
I will take a better look, however the Subscription model is not indexed by ES, and the value should be generated within the user session (if the user does not set a value, which is why it cannot be done on initialisation) so I don't think this is necessary. 

What I think I missed here is enforcing a value in the reference field, as the subscription mockups rely on having that value set. That would enforce why this needs to happen in a before hook as well.